### PR TITLE
New Metaphorpsum Spice

### PR DESCRIPTION
**Pull request by suhmantha1**
https://duck.co/ia/view/metaphorsum

### CODE REVIEW

**Comment by daxtheduck** [comment]
@suhmantha1 Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please head to https://quackslack.herokuapp.com/ to get an invite.

>More Info: http://docs.duckduckhack.com/submitting/submitting-overview.html

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by moollaza** [comment]
Hey @suhmantha1 I'm so sorry no on has reviewed this PR yet! It's currently deploying to Beta for testing. Let's get this reviewed and live!

File: share/spice/sentencer/sentencer.css [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.zci--sentencer {
```
Comment (moollaza) [inline share/spice/sentencer/sentencer.css]: Looks like you can remove this 👍

File: share/spice/sentencer/sentencer.handlebars [hunk 2]
```diff
@@ -0,0 +1 @@
+{{sam}}
```
Comment (moollaza) [inline share/spice/sentencer/sentencer.handlebars]: Looks like you can remove this too 👍

File: share/spice/sentencer/sentencer.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            },
+            meta: {
+                sourceName: "Metahorpsum",
+                sourceUrl: "http://metaphorpsum.com"
+            },
+
+            templates: {
+                group: 'text',
+                options: {
+                    content: Spice.sentencer.content,
```
Comment (moollaza) [inline share/spice/sentencer/sentencer.js]: You can remove this line as we don't need a custom template here

File: share/spice/sentencer/sentencer.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        
+        // Render the response
+        Spice.add({
+            id: "sentencer",
+            name: "Metahorpsum",
+            //data: api_result,
+            data: {
+                //content: api_result.text,
+                title: "Randomly generated metaphor",
+                subtitle: api_result.text
```
Comment (moollaza) [inline share/spice/sentencer/sentencer.js]: let's use `description` instead of `subtitle` here

**Comment by moollaza** [comment]
@suhmantha1 this is a great start! It looks like the API is pretty flexible too if we wanted to allow multiple paragraphs and sentences. I think we should at least split up the metaphors so its clear what they are?

I was thinking we could either use our list template, to produce a table with rows of metaphors, or we could mimic the design of the Lorem Ipsum Spice: https://duckduckgo.com/?q=3+paragraphs+of+lorem+ipsum&ia=answer&iax=1

Similarly, I think we can expand the triggering to allow for more natural queries like "3 sentences of metaphors", "4 random metaphors", and if you wanted, "5 paragraphs of metaphors" (this would need a different API call I think, unless the default call requests multiple paragraphs of single sentences?

What do you think?

**Comment by gaulrobe** [comment]
The package name and associated files are named as `Sentencer`. Should it be `Metaphorsum` to match the IA/source name?

File: lib/DDG/Spice/Sentencer.pm [hunk 5]
```diff
@@ -0,0 +1,32 @@
+package DDG::Spice::Sentencer;
+
+# ABSTRACT: Output a random number of metaphor sentences.
+
+use strict;
+use DDG::Spice;
+
+# spice proxy_cache_valid => "418 1d";
```
Comment (gaulrobe) [inline lib/DDG/Spice/Sentencer.pm]: I think we may want to uncomment this here... Users will want new results each time to mimic the API, so I don't think caching is necessary.

File: lib/DDG/Spice/Sentencer.pm [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+triggers startend => "metaphor", "metaphors", "metaphorpsum", "sentencer";
+
+spice to => 'http://metaphorpsum.com/sentences/$1?format=json';
+spice wrap_jsonp_callback => 1;
+
+handle remainder => sub {
+    #  print "SAAAAAMMMMMM!!!!!!";
+    #     print $_;
+    #     $_ = "$_";
+    #     print $_;
```
Comment (gaulrobe) [inline lib/DDG/Spice/Sentencer.pm]: This commented out code can be removed. While helpful for debugging, we probably don't want to check this in.

File: share/spice/sentencer/sentencer.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+            },
+            meta: {
+                sourceName: "Metahorpsum",
```
Comment (gaulrobe) [inline share/spice/sentencer/sentencer.js]: Typo here: `Metaphorpsum`

**Comment by moollaza** [comment]
@suhmantha1 Ping! I'm just checking in to see if you're still interested in finishing this one up? We'd love to see this go live!

There's absolutely no rush, just wanted to make sure you're not stuck on anything.

**Comment by suhmantha1** [comment]
@moollaza i will get to working on this again early next week!

**Comment by moollaza** [comment]
> @moollaza i will get to working on this again early next week!

Awesome 💯  Thanks a lot!

**Comment by pjhampton** [comment]
Hey @suhmantha1 

[DuckDuckHack is now in Maintenance Mode](https://duckduckhack.com/) and from now on, we are only accepting issues and PRs for essential bugs and bug fixes.

Unfortunately, this is non-essential and will be closed.

We appreciate you taking the time to contribute and apologize for not being able to include your work.

**Pull request closed without merging**